frontend/src/hooks/useScript.ts

```diff
@@ -0,0 +1,74 @@
+// Dynamically load an external script in one line with this React hook.
+// This can be useful to integrate a third party library like Google Analytics or Stripe.
+// This avoids loading this script in the <head> </head> on all your pages if it is not necessary.
+// Retrieved from https://usehooks-ts.com/react-hook/use-script.
+
+import { useEffect, useState } from 'react'
+
+type Status = 'idle' | 'loading' | 'ready' | 'error'
+
+export const useScript = (src?: string): Status => {
```

Comment:
I think the intention is to not load the script if nothing is supplied (i.e. dynamically choose to load scripts), and since `useScript` is a generic hook, we should keep that flexibility